### PR TITLE
Saas 11.5 new content seb

### PR DESCRIPTION
**Pull request by seb-odoo**
task-1885419

### CODE REVIEW

File: addons/website/static/src/js/menu/new_content.js [hunk 1]
```diff
@@ -1,29 +1,67 @@
 odoo.define('website.newMenu', function (require) {
 'use strict';
 
+var cleanUrl = require('website.helpers').cleanUrl;
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website/static/src/js/menu/new_content.js]: @qsm-odoo you don't need to check this one as I'm removing it

File: addons/website_slides/static/src/js/slides_upload.js [hunk 2]
```diff
@@ -378,4 +378,9 @@ $('.oe_slide_js_upload').on('click', function () {
     var channel_id = $(this).attr('channel_id');
     slides.page_widgets['upload_dialog'] = new SlideDialog(this, channel_id).appendTo(document.body);
 });
+
+if (window.location.href.includes('enable_slide_upload')) {
+    $('.oe_slide_js_upload').click();
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website_slides/static/src/js/slides_upload.js]: I added a comment

File: addons/website/static/src/js/menu/new_content.js [hunk 3]
```diff
@@ -65,23 +103,187 @@ var NewContentMenu = websiteNavbarData.WebsiteNavbarActionWidget.extend({
...
+            window.location.href = cleanUrl(window.location.href, true, true) + '?' + enableFlag;
+        }
+        function rpcFail() {
+            $i.removeClass()
+                .addClass('fa fa-exclamation-triangle');
+            $p.text(_.str.sprintf(_t('"%s" failed to install.'), name));
+        }
+
+        if (this.pendingInstall) {
+                content = _.str.sprintf(_t('The installation of a module is already in progress.'), name);
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website/static/src/js/menu/new_content.js]: I don't think we'll be able to translate the text while injecting the module name if we make it in xml? (not this one because it doesn't have a %s but the others?)

File: addons/website/models/ir_qweb.py [hunk 4]
```diff
@@ -37,6 +36,34 @@ class QWeb(models.AbstractModel):
     def get_asset_bundle(self, xmlid, files, remains=None, env=None):
         return AssetsBundleMultiWebsite(xmlid, files, remains=remains, env=env)
 
+    def _compile_directive_module(self, el, options):
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website/models/ir_qweb.py]: Doesn't it make the code much more complicated and less flexible?

```
<div groups="base.group_system" t-module="website_blog" class="col-md-4 mb8 js_new_element">
```
Becomes
```
<t t-set="website_blog" t-value="env['ir.module.module'].sudo().search([('name', '=', 'website_blog')])"/>
<div groups="base.group_system" t-att-data-module-id="website_blog.id" t-att-data-module-shortdesc="website_blog.shortdesc" class="col-md-4 mb8 js_new_element">
```

And

```
<xpath expr="//div[@id='o_new_content_menu_choices']//div[@t-module='website_blog']" position="attributes">
    <attribute name="t-module"/>
</xpath>
```
becomes
```
<xpath expr="//div[@id='o_new_content_menu_choices']//t[@t-set='website_blog']" position="replace"/>
<xpath expr="//div[@id='o_new_content_menu_choices']//div[@t-att-data-module-id='website_blog.id']" position="attributes">
    <attribute name="t-att-data-module-id"/>
    <attribute name="t-att-data-module-shortdesc"/>
    <attribute name="groups">website.group_website_designer</attribute>
</xpath>
```

[the remark this answers is not in the document]
Comment (JKE-be) [reply addons/website/models/ir_qweb.py]: Ne serais-tu pas de mauvaise fois sur ce coup ci ?

**Comment by seb-odoo** [comment]
Thanks for the review. I made all the requested changes. Can you check again please?

File: addons/website/static/src/js/menu/seo.js [hunk 5]
```diff
@@ -801,7 +799,7 @@ var SeoMenu = websiteNavbarData.WebsiteNavbarActionWidget.extend({
     init: function (parent, options) {
         this._super(parent, options);
 
-        if (window.location.href.includes('enable_seo')) {
+        if (window.location.search.includes('enable_seo')) {
```
Comment (JKE-be) [inline addons/website/static/src/js/menu/seo.js]: tu tiens vraiment a ton include ;) $.deparam

Comment (seb-odoo) [reply addons/website/static/src/js/menu/seo.js]: Changed, finally!

File: addons/website_slides/static/src/js/slides_upload.js [hunk 6]
```diff
@@ -378,4 +378,10 @@ $('.oe_slide_js_upload').on('click', function () {
     var channel_id = $(this).attr('channel_id');
     slides.page_widgets['upload_dialog'] = new SlideDialog(this, channel_id).appendTo(document.body);
 });
+
+// automatically open the upload dialog if requested from query string
+if (typeof $.deparam.querystring().enable_slide_upload !== 'undefined') {
```
Comment (qsm-odoo) [inline addons/website_slides/static/src/js/slides_upload.js]: !== undefined

File: addons/website_slides/static/src/js/website_slides.editor.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+                        var channel_url = this.$("option:selected").val();
+                        if (channel_url) {
+                            window.location.href = channel_url + '?enable_slide_upload';
+                        }
+                    }
+                }, {
+                    text: _t('Cancel'), close: true
+                },]
+            }).open();
+        });
```
Comment (qsm-odoo) [inline addons/website_slides/static/src/js/website_slides.editor.js]: Same: need to return a deferred here

File: addons/website_livechat/views/website_livechat.xml [hunk 8]
```diff
@@ -37,6 +37,13 @@
                 <link rel="stylesheet" type="text/scss" href="/im_livechat/static/src/scss/im_livechat.scss"/>
             </xpath>
         </template>
+
+        <template id="assets_editor_inherit_website_livechat" inherit_id="website.assets_editor" name="website_livechat Assets Editor">
```
Comment (qsm-odoo) [inline addons/website_livechat/views/website_livechat.xml]: Name those kind of template as the original one: `assets_editor`

File: addons/website_slides/views/website_slides.xml [hunk 9]
```diff
@@ -12,6 +12,12 @@
     </xpath>
 </template>
 
+<template id="assets_editor_inherit_website_slides" inherit_id="website.assets_editor" name="website_slides Assets Editor">
```
Comment (qsm-odoo) [inline addons/website_slides/views/website_slides.xml]: Idem

File: addons/website_slides/views/website_slides.xml [hunk 10]
```diff
@@ -99,7 +105,7 @@
                         </t>
                         <!-- Public user has not right to create/write on slide -->
                         <t t-if="channel.can_upload and not is_public_user">
-                            <a role="button" class="btn btn-primary oe_slide_js_upload" title="Upload Presentation" t-attf-channel_id="#{channel.id}" aria-label="Upload Presentation">
```
Comment (qsm-odoo) [inline addons/website_slides/views/website_slides.xml]: Those seem unrelated to the task (and commit)

File: addons/website/static/src/js/menu/new_content.js [hunk 11]
```diff
@@ -2,28 +2,72 @@ odoo.define('website.newMenu', function (require) {
...
-        'click > #o_new_content_menu_choices': '_onBackgroundClick',
+        'click': '_onBackgroundClick',
+        'keydown': '_onBackgroundKeydown',
     }),
+    // allow text to be customized with inheritance
+    newContentText: {
+        failed: _t('"%s" failed to install.'),
+        installInProgress: _t('The installation of a module is already in progress.'),
+        installNeeded: _t('You need to install the "%s" module in order to execute this action.'),
+        installPleaseWait: _t('Please wait while "%s" is being installed...'),
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website/static/src/js/menu/new_content.js]: I will change it when there is no double quote inside the string.

File: addons/website/static/src/js/menu/new_content.js [hunk 12]
```diff
@@ -2,28 +2,72 @@ odoo.define('website.newMenu', function (require) {
...
     events: _.extend({}, websiteNavbarData.WebsiteNavbarActionWidget.prototype.events || {}, {
-        'click > a': '_onMenuToggleClick',
-        'click > #o_new_content_menu_choices': '_onBackgroundClick',
+        'click': '_onBackgroundClick',
+        'keydown': '_onBackgroundKeydown',
     }),
+    // allow text to be customized with inheritance
+    newContentText: {
+        failed: _t('"%s" failed to install.'),
+        installInProgress: _t('The installation of a module is already in progress.'),
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website/static/src/js/menu/new_content.js]: I would prefer to keep "module" here as this is rather technical, and some of the modules we need to install are not shown with the "Apps" filter enabled on the backend (eg. website_livechat).

File: addons/website/static/src/js/menu/new_content.js [hunk 13]
```diff
@@ -2,28 +2,72 @@ odoo.define('website.newMenu', function (require) {
...
+     * Prepare the navigation and find the modules to install.
+     * Move not installed module buttons after installed modules buttons,
+     * but keep the original index to be able to move back the pending install
+     * button at its final position, so the user can click at the same place.
+     *
      * @override
      */
     start: function () {
+        var self = this;
+        var autoOpen = typeof $.deparam.querystring()[enableFlag] !== 'undefined';
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website/static/src/js/menu/new_content.js]: Ok to change

File: addons/website/static/src/js/menu/new_content.js [hunk 14]
```diff
@@ -2,28 +2,72 @@ odoo.define('website.newMenu', function (require) {
...
+
+        var $modules = this.$newContentMenuChoices.find('.js_new_element');
+        _.each($modules, function (el, index) {
+            var $el = $(el);
+            $el.data('original-index', index);
+            if ($el.data('module-id')) {
+
+                $el.appendTo($el.parent());
+                $el.find('a i').addClass('text-muted');
+                $el.find('a p').addClass('text-muted');
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website/static/src/js/menu/new_content.js]: I added the class on the i and p because the class needs to override custom css we put on them specifically. Should I try to improve the original css then?

File: addons/website/static/src/js/menu/new_content.js [hunk 15]
```diff
@@ -2,28 +2,72 @@ odoo.define('website.newMenu', function (require) {
...
+            }
+        });
+
+        this.$firstLink = this.$newContentMenuChoices.find('a:eq(0)');
+        this.$lastLink = this.$newContentMenuChoices.find('a:last');
+
+        return this._super.apply(this, arguments).then(function () {
+            if (autoOpen) {
+                self._onBackgroundClick();
+            }
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website/static/src/js/menu/new_content.js]: Ok for those changes

File: addons/website/static/src/js/menu/new_content.js (hunk 14 above)
[the remark this answers is not in the document]
Comment (qsm-odoo) [reply addons/website/static/src/js/menu/new_content.js]: No, but at least do this:
`$el.find('a i, a p').addClass('text-muted')`

File: addons/website/static/src/js/menu/new_content.js (hunk 12 above)
[the remark this answers is not in the document]
Comment (JKE-be) [reply addons/website/static/src/js/menu/new_content.js]: Fake
In this case, the new screen install Apps :)

[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website/static/src/js/menu/new_content.js]: Ok website_livechat is not an App but since it's auto-installed with im_livechat which is an App I guess I'll rename module into App.

File: addons/website/static/src/js/menu/new_content.js (hunk 14 above)
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website/static/src/js/menu/new_content.js]: ok

File: addons/website/static/src/js/menu/new_content.js [hunk 16]
```diff
@@ -65,23 +109,181 @@ var NewContentMenu = websiteNavbarData.WebsiteNavbarActionWidget.extend({
...
      *
      * @private
      * @param {Event} ev
      */
-    _onMenuToggleClick: function (ev) {
-        ev.preventDefault();
-        this.$newContentMenuChoices.toggleClass('o_hidden');
+    _onBackgroundClick: function (ev) {
+        // condition to prevent a crash when clicking on the button before the start is finished
+        if (this.$newContentMenuChoices) {
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website/static/src/js/menu/new_content.js]: Since we investigated it and it seems the issue is linked to my browser's debugger being messed up, I'll remove the condition.

File: addons/website/static/src/js/menu/new_content.js [hunk 17]
```diff
@@ -65,23 +109,181 @@ var NewContentMenu = websiteNavbarData.WebsiteNavbarActionWidget.extend({
...
+                    }
+                }
+                this.firstTab = false;
+                break;
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website/static/src/js/menu/new_content.js]: Ok

File: addons/website/static/src/js/menu/new_content.js [hunk 18]
```diff
@@ -65,23 +109,181 @@ var NewContentMenu = websiteNavbarData.WebsiteNavbarActionWidget.extend({
...
+        this.$lastLink.focus();
+    },
+    /**
+     * Hide the menu
+     *
+     * @private
+     */
+    _hideMenu: function () {
         this.$newContentMenuChoices.addClass('o_hidden');
+        $('body').removeClass('modal-open');
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website/static/src/js/menu/new_content.js]: So you prefer if I set the overflow: hidden; on the body manually in this case?

[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website/static/src/js/menu/new_content.js]: Or a custom class used only in this case? .new-content-open

File: addons/website/static/src/js/menu/new_content.js [hunk 19]
```diff
@@ -65,23 +109,181 @@ var NewContentMenu = websiteNavbarData.WebsiteNavbarActionWidget.extend({
...
+    },
+    /**
+     * Install a module
+     *
+     * @private
+     * @param {number} moduleId: the module to install
+     */
+    _install: function (moduleId) {
+        this.pendingInstall = 1;
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website/static/src/js/menu/new_content.js]: Good catch, I originally considered to allow queuing multiple install so I had to count it, before changing my mind. I'll change to boolean.

File: addons/website/static/src/js/menu/new_content.js [hunk 20]
```diff
@@ -65,23 +109,181 @@ var NewContentMenu = websiteNavbarData.WebsiteNavbarActionWidget.extend({
...
+    _hideMenu: function () {
         this.$newContentMenuChoices.addClass('o_hidden');
+        $('body').removeClass('modal-open');
+    },
+    /**
+     * Install a module
+     *
+     * @private
+     * @param {number} moduleId: the module to install
+     */
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website/static/src/js/menu/new_content.js]: Indeed I'll add the @return

File: addons/website/static/src/js/menu/new_content.js [hunk 21]
```diff
@@ -65,23 +109,181 @@ var NewContentMenu = websiteNavbarData.WebsiteNavbarActionWidget.extend({
...
+        var name = $el.data('module-shortdesc');
+
+        function rpcDone() {
+            window.location.href = window.location.origin + window.location.pathname + '?' + enableFlag;
+        }
+        function rpcFail() {
+            $i.removeClass()
+                .addClass('fa fa-exclamation-triangle');
+            $p.text(_.str.sprintf(self.newContentText.failed, name));
+        }
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website/static/src/js/menu/new_content.js]: Ok

File: addons/website/static/src/js/menu/new_content.js [hunk 22]
```diff
@@ -65,23 +109,181 @@ var NewContentMenu = websiteNavbarData.WebsiteNavbarActionWidget.extend({
...
+
+        if (this.pendingInstall) {
+                content = this.newContentText.installInProgress;
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website/static/src/js/menu/new_content.js]: Good catch

File: addons/website/static/src/js/menu/new_content.js [hunk 23]
```diff
@@ -65,23 +109,181 @@ var NewContentMenu = websiteNavbarData.WebsiteNavbarActionWidget.extend({
...
+        } else {
+            content = _.str.sprintf(this.newContentText.installNeeded, name);
+            buttons = [{
+                text: 'Install',
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website/static/src/js/menu/new_content.js]: 👍

File: addons/website/static/src/js/menu/new_content.js [hunk 24]
```diff
@@ -65,23 +109,181 @@ var NewContentMenu = websiteNavbarData.WebsiteNavbarActionWidget.extend({
...
+                classes: 'btn-primary',
+                close: true,
+                click: function () {
+                    // move the element where it will be after installation
+                    var finalPosition = self.$newContentMenuChoices
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website/static/src/js/menu/new_content.js]: Ok

File: addons/website/static/src/js/menu/seo.js [hunk 25]
```diff
@@ -217,13 +217,11 @@ var HtmlPage = Class.extend(mixins.PropertiesMixin, {
...
     url: function () {
-        var url = window.location.href;
-        var hashIndex = url.indexOf('?');
-        return hashIndex >= 0 ? url.substring(0, hashIndex) : url;
+        return window.location.origin + window.location.pathname;
     },
     title: function () {
         var $title = $('title');
-        return ($title.length > 0) && $title.text() && $title.text().trim();
+        return ($title.length > 0) && $title.text() && $title.text().trim() || '';
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website/static/src/js/menu/seo.js]: Ok

File: addons/website/static/src/js/menu/seo.js [hunk 26]
```diff
@@ -232,7 +230,7 @@ var HtmlPage = Class.extend(mixins.PropertiesMixin, {
     },
     description: function () {
         var $description = $('meta[name=description]');
-        return ($description.length > 0) && ($description.attr('content') && $description.attr('content').trim());
+        return ($description.length > 0) && ($description.attr('content') && $description.attr('content').trim()) || '';
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website/static/src/js/menu/seo.js]: Ok

File: addons/website/static/src/js/menu/seo.js [hunk 27]
```diff
@@ -801,7 +799,7 @@ var SeoMenu = websiteNavbarData.WebsiteNavbarActionWidget.extend({
     init: function (parent, options) {
         this._super(parent, options);
 
-        if (window.location.href.includes('enable_seo')) {
+        if (typeof $.deparam.querystring().enable_seo !== 'undefined') {
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website/static/src/js/menu/seo.js]: ok

File: addons/website/static/src/scss/website.ui.scss [hunk 28]
```diff
@@ -290,42 +290,32 @@ body .modal {
 #o_new_content_menu_choices {
     @include o-position-absolute($o-navbar-height, 0, 0, 0);
     position: fixed;
-    display: flex;
-    justify-content: center;
-    align-items: center;
     font-family: $o-we-font-family;
+    background-color: rgba(0, 0, 0, 0.8);
+    overflow: auto;
+    display: flex;
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website/static/src/scss/website.ui.scss]: Ok. Where can I find those guidelines?

File: addons/website/static/src/scss/website.ui.scss [hunk 29]
```diff
@@ -290,42 +290,32 @@ body .modal {
...
-        flex-flow: row wrap;
-        justify-content: space-around;
-        width: auto;
-        max-width: 60%;
-        max-height: 100%;
-        margin: 0;
-        padding: 0;
-        list-style: none;
+    .container {
+        max-width: 720px;
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website/static/src/scss/website.ui.scss]: Should I create a new class to give it the css?

File: addons/website/static/src/scss/website.ui.scss [hunk 30]
```diff
@@ -290,42 +290,32 @@ body .modal {
...
+        margin: auto;
+    }
 
-        li {
-            display: block;
+    > div {
+
+        > div {
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website/static/src/scss/website.ui.scss]: Ok, I even created that class for another purpose.

File: addons/website_livechat/static/src/js/website_livechat.editor.js [hunk 31]
```diff
@@ -0,0 +1,40 @@
...
+                args: [[]],
+                kwargs: {
+                    context: weContext.get(),
+                    name: name,
+                },
+            }).then(function (url) {
+                window.location.href = url;
+            });
+        });
+    },
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website_livechat/static/src/js/website_livechat.editor.js]: Ok

File: addons/website_slides/static/src/js/slides_upload.js (hunk 6 above)
Comment (seb-odoo) [reply addons/website_slides/static/src/js/slides_upload.js]: ok

File: addons/website_slides/static/src/js/website_slides.editor.js (hunk 7 above)
Comment (seb-odoo) [reply addons/website_slides/static/src/js/website_slides.editor.js]: ok

File: addons/website_livechat/views/website_livechat.xml (hunk 8 above)
Comment (seb-odoo) [reply addons/website_livechat/views/website_livechat.xml]: Just for the id or also for the name?

File: addons/website_slides/views/website_slides.xml (hunk 9 above)
Comment (seb-odoo) [reply addons/website_slides/views/website_slides.xml]: ok

File: addons/website_slides/views/website_slides.xml (hunk 10 above)
Comment (seb-odoo) [reply addons/website_slides/views/website_slides.xml]: Indeed, I ended up doing them on the other PR, they should disappear at the rebase (if merged second).

File: addons/website/static/src/js/menu/new_content.js (hunk 18 above)
[the remark this answers is not in the document]
Comment (qsm-odoo) [reply addons/website/static/src/js/menu/new_content.js]: Custom class I suppose yes

File: addons/website/static/src/scss/website.ui.scss (hunk 28 above)
[the remark this answers is not in the document]
Comment (qsm-odoo) [reply addons/website/static/src/scss/website.ui.scss]: https://github.com/odoo/odoo/wiki/SCSS-coding-guidelines

File: addons/website/static/src/scss/website.ui.scss (hunk 29 above)
[the remark this answers is not in the document]
Comment (qsm-odoo) [reply addons/website/static/src/scss/website.ui.scss]: Bootstrap should have a standard layout which is close enough to what you want, to avoid adding more style of ours. But it is fine like this for now.

File: addons/website/static/src/js/menu/new_content.js [hunk 32]
```diff
@@ -65,23 +109,181 @@ var NewContentMenu = websiteNavbarData.WebsiteNavbarActionWidget.extend({
...
+                        this._focusLastLink();
+                        ev.preventDefault();
+                    }
+                } else {
+                    if (this.firstTab || document.activeElement === this.$lastLink[0]) {
+                        this._focusFirstLink();
+                        ev.preventDefault();
+                    }
+                }
+                this.firstTab = false;
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/website/static/src/js/menu/new_content.js]: As discussed, we want to capture the navigation inside the popup, because it doesn't make sense to be able to select elements that are potentially "hidden" below. This is the same behavior as bs4 modal.

If you decided to click on "new" and wanted to navigate to "edit" instead, you can just press esc first and then tab.

File: addons/website_livechat/views/website_livechat.xml (hunk 8 above)
Comment (seb-odoo) [reply addons/website_livechat/views/website_livechat.xml]: Not done, to follow guideline.

File: addons/website_slides/views/website_slides.xml (hunk 9 above)
Comment (seb-odoo) [reply addons/website_slides/views/website_slides.xml]: Not done, to follow guideline.

**Comment by seb-odoo** [comment]
@qsm-odoo changes made, can you check again please? Especially the actions/deferred

File: addons/website/static/src/js/menu/new_content.js [hunk 33]
```diff
@@ -56,32 +92,192 @@ var NewContentMenu = websiteNavbarData.WebsiteNavbarActionWidget.extend({
                 var url = '/website/add/' + encodeURIComponent(val);
                 if ($dialog.find('input[type="checkbox"]').is(':checked')) url +='?add_menu=1';
                 document.location = url;
+            } else {
+                def.reject();
             }
+        }, def.resolve.bind(def));
+        return def;
```
Comment (qsm-odoo) [inline addons/website/static/src/js/menu/new_content.js]: Same as in the other PR (for all ones)

Comment (qsm-odoo) [reply addons/website/static/src/js/menu/new_content.js]: (ok for the rest)

File: addons/website/views/website_navbar_templates.xml [hunk 34]
```diff
@@ -109,9 +109,46 @@
...
+                                        </a>
+                                    </div>
+                                    <div groups="base.group_system" name="module_website_forum" t-att-data-module-id="env.ref('base.module_website_forum').id" t-att-data-module-shortdesc="env.ref('base.module_website_forum').shortdesc" class="col-md-4 mb8 o_new_content_element">
+                                        <a href="#" data-action="new_forum">
+                                            <i class="fa fa-comment"/>
+                                            <p>New Forum</p>
+                                        </a>
+                                    </div>
+                                    <div groups="base.group_system" name="module_website_hr_recruitment" t-att-data-module-id="env.ref('base.module_website_hr_recruitment').id" t-att-data-module-shortdesc="env.ref('base.module_website_hr_recruitment').shortdesc" class="col-md-4 mb8 o_new_content_element">
+                                        <a href="/jobs/add">
```
Comment (FalcoBolger) [inline addons/website/views/website_navbar_templates.xml]: @seb-odoo shouldn't it be `href="#"`?

Comment (JKE-be) [reply addons/website/views/website_navbar_templates.xml]: why would you like to have smthing that do nothing ? 

Most of other object uses click binding to create their new record. $('.new_forum').on('click', fun())
In case of job, the controller /job/add do it without js, so a direct link seems legits.

Do you have a bug ?

Comment (seb-odoo) [reply addons/website/views/website_navbar_templates.xml]: @FalcoBolger It was question of changing it to # in website but with an override in website_job to the correct url (because the url doesn't exist in website itself, there was an issue with an url crawler or something like that).

But for the feature itself as @JKE-be said, the other actions are handled in JS but this one is just a link so it should be different in that regard.

Comment (FalcoBolger) [reply addons/website/views/website_navbar_templates.xml]: @JKE-be @seb-odoo the crawler fails when the website_hr_recruitment module is not installed, so it seems better to have the module itself adding the link to the endpoint.

Just a suggestion, nothing that an inherited template could not fix.

Comment (seb-odoo) [reply addons/website/views/website_navbar_templates.xml]: See PR #29217
